ci/cifailurejob.yaml

```diff
@@ -0,0 +1,166 @@
+                        }
+                    }
+                }
+                def waitForStageToComplete(build_id,stage_number)
+                {
+                    echo "build_id: ${build_id} stage: ${stage_number}"
+                    waitUntil{
+                        build_status = sh(returnStdout: true, script: "oc get builds ${build_id} -o template  --template='{{index .metadata.annotations \"openshift.io/jenkins-status-json\"}}'").trim()
+                        def build_status_obj = new JsonSlurperClassic().parseText(build_status)
+                        if ( build_status_obj.stages[stage_number] == null ){
+                            sleep 30
+                            false
+                        }
+                        else{
+                            if ( build_status_obj.stages[stage_number]['status'] != 'SUCCESS' && build_status_obj.stages[stage_number]['status'] != 'FAILED')
+                            {
+                                echo "Stage status is: ${build_status_obj.stages[stage_number]['status']} waiting to complete"
+                                false
+                            }
+                            else
+                            {
+                                echo "Stage complete"
+                                true
+                            }
+                        }
+                    }
+                }
+                def checkStageStatus(build_id,stage_number,expected_status)
+                {
+                    echo "build_id: ${build_id} stage: ${stage_number} expected_status: ${expected_status}"
+                    def actual_status = false
+
+                    if ( expected_status != "not_available" )
+                    {
+                        waitForStageToComplete(build_id,stage_number)
+
+                        build_status = sh(returnStdout: true, script: "oc get builds ${build_id} -o template  --template='{{index .metadata.annotations \"openshift.io/jenkins-status-json\"}}'").trim()
+                        def build_status_obj = new JsonSlurperClassic().parseText(build_status)
+                        def stage_name = build_status_obj.stages[stage_number]['name']
+                        def stage_status = build_status_obj.stages[stage_number]['status']
+                        echo "Stage name: ${stage_name}   Stage status: ${stage_status}"
+                        if ( stage_status.toLowerCase() == expected_status.toLowerCase() )
+                        {
+                            echo "Stage assertion passed"
+                            actual_status = true
+                        }
+                        else
+                        {
+                            echo "Stage assertion failed"
+                        }
+                    }
+                    else
+                    {
+                        echo "The stage $stage_number, should not be present in the pipeline, checking for it"
```

Comment:
Isn't it more readable to read the stage name rather than stage number? Can we also add stage name to this statement?

Reply:
Here we are checking for a stage, which ideally should be non-existent. So we do not know the name. This is only for our debugging so that we come to know, there is some random stage is running.